google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -182,13 +181,10 @@ def __init__(
         self._auth_init()
         self._priv_key, pub_key = generate_keys()
         self._pub_key = pub_key.decode("UTF-8")
-        self._lock = threading.Lock()
 
         logger.debug("Updating instance data")
 
-        with self._lock:
-            self._current = self._perform_refresh()
-            self._next = self.immediate_future(self._current)
+        asyncio.run_coroutine_threadsafe(self._perform_refresh(), self._loop).result()
```

Comment:
So. we don't want to do this, because this means that we're blocking on the refresh for the initialization to complete. 

Does this work? 
```
self._current = self._perform_refresh()
self._next = self._current
asyncio.run_coroutine_threadsafe(self._current, self._loop)
```

Reply:
Fixed